src/Fixer/AbstractPhpUnitFixer.php

```diff
@@ -116,26 +126,51 @@ private function createDocBlock(Tokens $tokens, int $docBlockIndex, string $anno
 
     /**
      * @param list<string> $preventingAnnotations
+     * @param list<string> $preventingAttributes
      */
     private function updateDocBlockIfNeeded(
         Tokens $tokens,
         int $docBlockIndex,
         string $annotation,
-        array $preventingAnnotations
+        array $preventingAnnotations,
+        array $preventingAttributes
     ): void {
         $doc = new DocBlock($tokens[$docBlockIndex]->getContent());
         foreach ($preventingAnnotations as $preventingAnnotation) {
             if ([] !== $doc->getAnnotationsOfType($preventingAnnotation)) {
                 return;
             }
         }
+
+        if ($this->isPreventedByAttribute($tokens, $docBlockIndex, $preventingAttributes)) {
+            return;
+        }
+
         $doc = $this->makeDocBlockMultiLineIfNeeded($doc, $tokens, $docBlockIndex, $annotation);
         $lines = $this->addInternalAnnotation($doc, $tokens, $docBlockIndex, $annotation);
         $lines = implode('', $lines);
 
         $tokens[$docBlockIndex] = new Token([T_DOC_COMMENT, $lines]);
     }
 
+    /**
+     * @param list<string> $preventingAttributes
+     */
+    private function isPreventedByAttribute(Tokens $tokens, int $docBlockIndex, array $preventingAttributes): bool
+    {
+        $classIndex = $tokens->getNextTokenOfKind($docBlockIndex, [[T_CLASS]]);
+
+        foreach (AttributeAnalyzer::getAttributesForElement($tokens, $classIndex) as $attributeAnalysis) {
+            $attributeNameIndex = $tokens->getNextMeaningfulToken($attributeAnalysis->getStartIndex());
+            $attributeName = $tokens[$attributeNameIndex]->getContent();
```

Comment:
I believe this is too naive approach, which will work only for short names (`#[CoversClass]`), but won't work for `#[\PHPUnit\Framework\Attributes\CoversClass()]` or `#[PHPUnitAttributes\CoversClass]` when imported with `use PHPUnit\Framework\Attributes as PHPUnitAttributes;`. I believe that fixer requires full name extracted from the attribute (which #7909 does), and on top of that needs to make FQN resolution based on existing imports. We need to make sure that attribute is exactly the PHPUnit's one 🙂.

Reply:
> I believe this is too naive approach

Is it? Let's try to be practical with a starting implementation. Anyone having `#[PHPUnitAttributes\CoversClass]` who uses a fixer to have some consistency in code should care about having it converted to `#[CoversClass]` in the first place, right? Or am I crazy hoping for that?

Reply:
Standards are configurable, you can have FQCN with leading backslash too if you want. If anyone would want an aliased relative FQN, then it's totally valid too. The thing is that currently this fixer treats an attribute just like it was an annotation (simple string), while it's not. `CoversClass` is only part of an identifier, the rest is imported. It would skip if `Foo\CoversClass` used, which is wrong IMHO.